db/model/Gdoc/GdocAuthor.ts

```diff
@@ -25,6 +29,32 @@ export class GdocAuthor extends GdocBase implements OwidGdocAuthorInterface {
         return gdoc.content.bio ?? []
     }
 
+    _loadSubclassAttachments = async (): Promise<void> => {
+        if (!this.content.title) return
+        const knex = db.knexInstance()
+
+        this.latestWorkLinks = await getLatestWorkByAuthor(
+            knex,
+            this.content.title
+        )
+        if (!this.latestWorkLinks) return
+
+        // We want to load additional image filenames from the referenced
+        // latest work links. We're not relying here on the Link
+        // infrastructure as we don't yet have a good way of cleaning up
+        // obsolete links from the latest work section. Usually the links
+        // originating from a gdoc are cleaned up when updating/deleting it, but
+        // here the links from the latest section will change independently of a
+        // manual authoring and admin action, so they'll end up being stale
+        // until the author page is updated again.
+        const latestWorkImageFilenames = this.latestWorkLinks
+            .map((d) => d["featured-image"])
+            .filter(Boolean)
+
+        // Load the image metadata for the latest work images
+        return super.loadImageMetadata(latestWorkImageFilenames)
```

Comment:
@ikesau I'd be curious to hear whether this adjustment to parsing images in a block, whose content is dynamically generated, fits your mental model of image handling.

Reply:
I don't understand this comment - can you give me a bit of context for this one? Is this a concern because of lightning baking? Why can we not just iterate over the latest work and collect the featured-images?

Reply:
> Why can we not just iterate over the latest work and collect the featured-images?

@danyx23 I'm assuming you're referring to https://github.com/owid/owid-grapher/blob/904b82cc96cc6a0857486d025b69b028731fa9c2/db/model/Gdoc/GdocBase.ts#L276-L286

In this snippet `this.linkedDocuments` originate from `this.links`, which is populated in parallel to `Link.createFromUrl` calls, which create rows in the `posts_gdocs_links` table.

These links only get refreshed when updating/deleting a gdoc in the admin: https://github.com/owid/owid-grapher/blob/904b82cc96cc6a0857486d025b69b028731fa9c2/adminSiteServer/apiRouter.ts#L2465-L2469

The problem is that there won't be an admin action for every update to the latest work block, which is updated dynamically whenever a new article by a given author is published/unpublished. So the `posts_gdocs_links` table will gradually fall out of sync.

By contrast, the featured work block is manually updated from content in the gdoc itself, so modifying the block necessarily updates the `posts_gdocs_links` table.